driver-pulsar/src/main/java/io/nosqlbench/driver/pulsar/ops/PulsarAdminTopicOp.java

```diff
@@ -0,0 +1,148 @@
+    public void run() {
+        PulsarAdmin pulsarAdmin = clientSpace.getPulsarAdmin();
+        Topics topics = pulsarAdmin.topics();
+
+        try {
+            // Create the topic
+            if (!adminDelOp) {
+                if (!partitionTopic) {
+                    if (!asyncApi) {
+                        topics.createNonPartitionedTopic(topicUri);
+                        logger.trace("Successfully created non-partitioned topic \"" + topicUri + "\" synchronously!");
+                    } else {
+                        CompletableFuture<Void> future = topics.createNonPartitionedTopicAsync(topicUri);
+                        future.whenComplete((unused, throwable)
+                            -> logger.trace("Successfully created non-partitioned topic \"" + topicUri + "\" asynchronously!"))
+                        .exceptionally(ex -> {
+                            logger.error("Failed to create non-partitioned topic \"" + topicUri + "\" asynchronously!");
+                            return null;
+                        });
+                    }
+                } else {
+                    if (!asyncApi) {
+                        topics.createPartitionedTopic(topicUri, partitionNum);
+                        logger.trace("Successfully created partitioned topic \"" + topicUri + "\"" +
+                            "(partition_num: " + partitionNum + ") synchronously!");
+                    } else {
+                        CompletableFuture<Void> future = topics.createPartitionedTopicAsync(topicUri, partitionNum);
+                        future.whenComplete((unused, throwable)
+                            -> logger.trace("Successfully created partitioned topic \"" + topicUri + "\"" +
+                            "(partition_num: " + partitionNum + ") asynchronously!"))
+                        .exceptionally(ex -> {
+                            logger.error("Failed to create partitioned topic \"" + topicUri + "\"" +
+                                "(partition_num: " + partitionNum + ") asynchronously!");
+                            return null;
+                        });
+                    }
+                }
+            }
+            // Delete the topic
+            else {
+                if (!partitionTopic) {
+                    if (!asyncApi) {
+                        topics.delete(topicUri, true);
+                        logger.trace("Successfully deleted non-partitioned topic \"" + topicUri + "\" synchronously!");
+                    } else {
+                        CompletableFuture<Void> future = topics.deleteAsync(topicUri, true);
+                        future.whenComplete((unused, throwable)
+                            -> logger.trace("Successfully deleted non-partitioned topic \"" + topicUri + "\" asynchronously!"))
+                        .exceptionally(ex -> {
+                            logger.error("Failed to delete non-partitioned topic \"" + topicUri + "\" asynchronously!");
```

Comment:
In OulsarProducerOp I introduced a framework to fail in case of async failure.
Can you please take a look?

Reply:
Thanks for the review. I'll take a look.